engine/runtime/src/main/scala/org/enso/compiler/context/ChangesetBuilder.scala

```diff
@@ -42,8 +58,58 @@ final class ChangesetBuilder[A: TextEditor: IndexedSource](
     * @return the computed changeset
     */
   @throws[CompilerError]
-  def build(edits: Seq[TextEdit]): Changeset[A] =
-    Changeset(source, ir, compute(edits))
+  def build(edits: Seq[TextEdit]): Changeset[A] = {
+
+    val simpleEditOption: Option[TextEdit] = edits match {
+      case Seq(first) => Some(first)
+      case Seq(head, realEdit) => {
+        val firstAffected = invalidated(Seq(head))
+        if (firstAffected.isEmpty) {
+          Some(realEdit)
+        } else {
+          None
+        }
+      }
+      case _ => None
+    };
+
+    val simpleUpdateOption = simpleEditOption
+      .filter(edit => edit.range.start.line == edit.range.end.line)
+      .map(edit => (edit, invalidated(Seq(edit))))
+      .filter({ case (_, directlyAffected) => directlyAffected.size == 1 })
+      .flatMap({ case (edit, directlyAffected) =>
```

Comment:
```suggestion
      .flatMap { case (edit, directlyAffected) =>
```
together with the other `}`